src/main/cli/cli.c

```diff
@@ -1315,30 +1316,47 @@ static void cliSerial(const char *cmdName, char *cmdline)
 
     uint8_t validArgumentCount = 0;
 
-    const char *ptr = cmdline;
+    char *ptr = cmdline;
+    char *tok = strsep(&ptr, " ");
+    serialPortIdentifier_e identifier = findSerialPortByName(tok, strcasecmp);
+    if (identifier == SERIAL_PORT_NONE) {
+        char *eptr;
+        identifier = strtoul(tok, &eptr, 10);
+        if (*eptr) {
+            // parsing ended before end of token indicating an invalid identifier
+            identifier = SERIAL_PORT_NONE;
+        } else {
+           // when using index then UART1 == 0
+           if (identifier >= SERIAL_PORT_START_INDEX && identifier < SERIAL_UART_MAX - 1 + SERIAL_UART_FIRST_INDEX) {
```

Comment:
it is starting from zero (uart 1 before this pr). SERIAL_PORT_START_INDEX may be changed later

Reply:
Agree but my point is if we can't assume NONE == -1 we should also not assume UART0/1 == 0

Reply:
this is backward compatibility code. It is possible to name old uart1, buy not strictly necessary. This code should be removed in future, when everyone migrates to names.

in SITL case, identifier is used directly as index into array. And array starts from zero.